### PR TITLE
Spark 3.3.2 fork with K8s / Iceberg support

### PR DESCRIPTION
**Pull request by rayhondo**
First release of spark 3.3.2 for Affirm. This mainly amends the official spark release to:
- use specific build command for Affirm's spark k8s distribution
- adds jars in `pom.xml` file to be packaged in distribution
- sets up spark k8s entrypoint in build

### CODE REVIEW

**Review by chadlagore — APPROVED** [review]
I like the branching strategy for versions, please give your thoughts @andrewdanks

**Comment by rayhondo** [comment]
> I like the branching strategy for versions, please give your thoughts @andrewdanks

[spark versioning strategy draft](https://docs.google.com/document/d/1sJ9Ou1uwqImuvwYORlPh0NymVZPjgald5gI-7Zc3g9c/edit)

File: python/setup.py [hunk 1]
```diff
@@ -77,14 +77,15 @@
 USER_SCRIPTS_PATH = os.path.join(SPARK_HOME, "sbin")
 DATA_PATH = os.path.join(SPARK_HOME, "data")
 LICENSES_PATH = os.path.join(SPARK_HOME, "licenses")
+DOCKER_PATH = os.path.join(SPARK_HOME, "resource-managers/kubernetes/docker/src/main/dockerfiles/spark")
```
Comment (michael-sun-affirm) [inline python/setup.py:80]: Why it's needed?

File: resource-managers/kubernetes/core/pom.xml [hunk 2]
```diff
@@ -95,6 +95,22 @@
       </exclusions>
     </dependency>
 
+    <dependency>
+      <groupId>io.kubernetes</groupId>
```
Comment (michael-sun-affirm) [inline resource-managers/kubernetes/core/pom.xml:99]: A bit surprising that this lib is not included in the official build.

File: python/setup.py (hunk 1 above)
Comment (rayhondo) [reply python/setup.py:80]: These files are copied into the build as `deps/k8s` and the `entrypoint.sh` is called in the Affirm batch Dockerfile 

https://github.com/Affirm/all-the-things/blob/87ab4991fd07115c6f97fbceb735ab13510a15a9/deployable/batch-infra-tasks/src/Dockerfile#L9-L10

**Pull request merged**